### PR TITLE
BindingStep fixes enabled by Workflow 1.5

### PR DESCRIPTION
**Pull request by jglick**
- [X] [JENKINS-27389](https://issues.jenkins-ci.org/browse/JENKINS-27389) testing `env.SECRETVAR` effect of https://github.com/jenkinsci/workflow-plugin/pull/41.
- [X] [JENKINS-27631](https://issues.jenkins-ci.org/browse/JENKINS-27631) security fix, based on https://github.com/jenkinsci/workflow-plugin/pull/106.
- [X] [JENKINS-27486](https://issues.jenkins-ci.org/browse/JENKINS-27486) mask secrets accidentally printed to console, based on https://github.com/jenkinsci/workflow-plugin/pull/107.

@reviewbybees

### CODE REVIEW

**Comment by tfennelly** [comment]
:+1:

File: pom.xml [hunk 1]
```diff
@@ -61,25 +61,25 @@
...
-      <version>${workflow.version}</version>
+      <version>1.5-20150331.125207-5</version>
       <classifier>tests</classifier>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins.workflow</groupId>
       <artifactId>workflow-support</artifactId>
-      <version>${workflow.version}</version>
+      <version>1.5-20150331.125230-5</version>
```
Comment (kohsuke) [inline pom.xml]: So I guess multi-module Maven deployment can't use the same timestamp? ouch.

File: src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java [hunk 2]
```diff
@@ -193,6 +194,52 @@
...
+    @Test public void masking() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                String credentialsId = "creds";
+                String secret = "s3cr3t";
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), new StringCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, "sample", Secret.fromString(secret)));
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  withCredentials([[$class: 'StringBinding', credentialsId: '" + credentialsId + "', variable: 'SECRET']]) {\n"
```
Comment (kohsuke) [inline src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java]: `$class` must die, not in this PR, but in a not so distant future.

**Comment by kohsuke** [comment]
:+1:

File: pom.xml (hunk 1 above)
Comment (jglick) [inline pom.xml]: Correct. Ouch indeed. I have spent the past several days redeploying snapshots and juggling branches.

**Pull request merged**